agent/types/types.go

```diff
@@ -0,0 +1,34 @@
+// This package is the contract between the agent and the SDK. It allows to
+// strictly separate the SDK from the agent package since the agent does not
+// export its internals.
+
+package types
+
+import (
+	"net/http"
+	"time"
+)
+
+type Agent interface {
+	NewRequestRecord(req *http.Request) RequestRecord
+	GracefulStop()
+}
+
+type RequestRecord interface {
```

Comment:
I would not expose the concept of Request Record and rather expose that as SDK

Reply:
I need a name for the structure we attach to HTTP contextes. NewRelic called their structure `Transaction`, and for us I chose "request record of of events", ie. `HTTPRequestRecord`.

There is only one type exposed to users and it is `HTTPRequestRecord`. The others are exported by the agent but meant to be used by the SDK, and indeed there are no way to use them is the SDK interface.